thetis/shallowwater_eq.py

```diff
@@ -854,9 +893,16 @@ def __init__(self, eta_test, eta_space, u_space,
         :arg bathymetry: bathymetry of the domain
         :type bathymetry: :class:`Function` or :class:`Constant`
         :kwarg bool nonlin: If False defines the linear shallow water equations
+        :kwarg wd_alpha: wetting-drying parameter
         """
-        super(FreeSurfaceEquation, self).__init__(eta_space, bathymetry, nonlin)
-        self.add_continuity_terms(eta_test, eta_space, u_space, bathymetry, nonlin)
+        super(FreeSurfaceEquation, self).__init__(eta_space, bathymetry, nonlin, wd_alpha)
+        self.add_continuity_terms(eta_test, eta_space, u_space, bathymetry, nonlin, wd_alpha)
+        self.bathymetry_displacement_mass_term = BathymetryDisplacementMassTerm(eta_test, eta_space, u_space, bathymetry, nonlin, wd_alpha)
+
+    def mass_term(self, solution):
+        f = super(ShallowWaterEquations, self).mass_term(solution)
+        f += self.bathymetry_displacement_mass_term.residual(solution)
```

Comment:
A bug here, currently the correct argument is `eta`. Would be cleaner to comply with the standard.